src/fhir/models/diagnostic-report.ts

```diff
@@ -67,7 +100,21 @@ export class DiagnosticReportModel extends FHIRModel<fhir4.DiagnosticReport> {
   }
 
   get details() {
-    return this.results.length > 0 ? `${this.results.length} results available` : "";
+    let display = "";
+    switch (this.results.length) {
+      case 0:
+        return "";
+      case 1:
+        display = `${this.results.length} result`;
+        break;
+      default:
+        display = `${this.results.length} results`;
+        break;
+    }
+    if (this.trends) {
+      display += ", result trend available";
+    }
+    return display;
```

Comment:
Up to you, might read better more compact
```suggestion
    if (!this.results.length) {
      return "";
    }
    const tResult = this.results.length > 1 ? "results" : "result";
    let display = `${this.results.length} ${tResult}`;
    if (this.trends) {
      display += ", result trend available";
    }
    return display;
```